### PR TITLE
Changes to tests to use the record helper class and delete old ones

### PR DESCRIPTION
**Pull request by alfredomusumeci**
- A Result record class has been created
- GeneratorTest and CustomGenerationTest now both use the Result class

### CODE REVIEW

File: src/test/java/com/gs/crdtools/BUILD [hunk 1]
```diff
@@ -18,6 +18,7 @@ java_junit5_test(
         "//src/main/java/com/gs/crdtools:openapi-src-genner",
         "//src/main/java/com/gs/crdtools:spec-extractor",
         "//src/main/java/com/gs/crdtools/codegen:swagger-codegen-extensions",
+        "//src/test/java/com/gs/crdtools/codegen:tests",
```
Comment (nresare) [inline src/test/java/com/gs/crdtools/BUILD:21]: Avoid having a less specific module depend on a more specific one. I would put the Result record definition in a `Results.java` in test/java/com/gs/crdtools, add it to the list of sources in the tests target and depend on that target in the codegen tests

File: src/test/java/com/gs/crdtools/GeneratorTest.java [hunk 2]
```diff
@@ -36,14 +36,14 @@ void testGenerate() throws IOException {
 
         var input = Path.of(runFiles.rlocation("__main__/src/test/resources/minimal-crd.yaml"));
         var parsedCrd = Generator.parseCrds(List.of(input));
-        var sourceCodeFromSpecs = Generator.generate(parsedCrd);
+        var sourceCodeFromSpecs = new CustomGenerationTest.Result(Generator.generate(parsedCrd));
```
Comment (nresare) [inline src/test/java/com/gs/crdtools/GeneratorTest.java:39]: If we move the Result definition to a separate file this will solve itself, but otherwise I think this is an excellent example of where it makes sense to use a static import and get rid of the surrounding class

File: src/test/resources/BUILD [hunk 3]
```diff
@@ -1,6 +1,6 @@
 filegroup(
     name = "all",
-    srcs = glob(["*.txt"]) + ["minimal-openapi.yaml"],
```
Comment (nresare) [inline src/test/resources/BUILD:3]: removing those two files is unrelated to the other change. Ideally it should go in a separate commit

**Review by nresare — COMMENTED** [review]
I like the ideas, some feedback below though

File: src/test/java/com/gs/crdtools/BUILD (hunk 1 above)
Comment (alfredomusumeci) [reply src/test/java/com/gs/crdtools/BUILD:21]: I was thinking about this, but I remembered your words about creating utility classes only when there is actually more than one function that we are using, hence why I did not do that. My understanding now is that this rule can be _broken_ when there is a situation like this of a less specific module being dependent on a more specific one. Understood!

**Comment by alfredomusumeci** [comment]
Updated as discussed

**Review by nresare — APPROVED** [review]
